app/models/financial_transaction.rb

```diff
@@ -5,11 +5,14 @@ class FinancialTransaction < ApplicationRecord
   belongs_to :user
   belongs_to :financial_link
   belongs_to :financial_transaction_type
+  belongs_to :reverts, class_name: 'FinancialTransaction', foreign_key: 'reverts'
+  has_one :reverted_by, class_name: 'FinancialTransaction', foreign_key: 'reverts'
```

Comment:
What about naming this foreign key `reverts_id` in the database?

Reply:
done